lmfit/model.py

```diff
@@ -56,6 +56,77 @@ def no_op(*args, **kwargs):
     return no_op
 
 
+def get_reducer(option):
+    """Factory function to build a parser for complex numbers.
+    `option` should be one of `['real', 'imag', 'abs', 'angle']`"""
+    assert option in ['real', 'imag', 'abs', 'angle'], "Unsupported option!"
+
+    def reducer(array):
+            """Convert a complex array to a real array based on the option passed
+            to parse_complex. Does nothing to a purely real array."""
+            if any(np.iscomplex(array)):
+                parsed_array = getattr(np, option)(array)
+            else:
+                parsed_array = array
+
+            return parsed_array
+    return reducer
+
+
+def propagate_err(z, dz, option):
+    """Perform error propagation on a vector of complex uncertainties to
+    get values for magnitude (abs) and phase (angle) uncertainty.
+    Uncertainties are 1/weights. If the weights provided are real, they are
+    assumed to apply equally to the real and imaginary parts. If the weights
+    are complex, the real part of the weights are applied to the real
+    part of the residual and the imaginary part is treated correspondingly."""
```

Comment:
minor detail: the docstrings don't completely follow the conventions (https://www.python.org/dev/peps/pep-0257/)

Reply:
OK, I updated the docstrings quite a bit. Check them out and see if they pass muster.